test/e2e/mock-response-data/snaps/snap-binary-mocks.ts

```diff
@@ -0,0 +1,328 @@
+import fs from 'fs';
+import { escapeRegExp } from 'lodash';
+import { Mockttp } from 'mockttp';
+
+const SNAP_ASSETS_RELATIVE_PATH =
+  'test/e2e/mock-response-data/snaps/snap-binaries-and-headers';
+const NPM_REGISTRY_METAMASK_BASE_URL = 'https://registry.npmjs.org/@metamask';
+
+/**
+ * Sorts an array of file names by version strings semantically (e.g., 1.2.0, 1.10.0, 2.0.0).
+ * Modifies the array in place.
+ *
+ * @param versions - An array of version strings to sort.
+ */
+function sortFilesByVersion(versions: string[]): void {
+  versions.sort((a, b) => {
+    const partsA = a.split('.').map(Number);
+    const partsB = b.split('.').map(Number);
+    for (let i = 0; i < Math.max(partsA.length, partsB.length); i++) {
+      const valA = partsA[i] ?? 0;
+      const valB = partsB[i] ?? 0;
+      if (valA < valB) {
+        return -1;
+      }
+      if (valA > valB) {
+        return 1;
+      }
+    }
+    return 0;
+  });
+}
+
+/**
+ * Retrieves the latest version of a snap from the local file system based on its name prefix.
+ * It looks for files named `<snapNamePrefix>-<version>.txt`, collects all versions,
+ * sorts them semantically, and returns the latest one.
+ *
+ * @param snapNamePrefix - The prefix of the snap name (e.g., 'bip32-example-snap').
+ * @returns The latest version string found for the snap.
+ * @throws If no version file is found for the given snap prefix in the directory.
+ */
+export function getLocalSnapLatestVersion(snapNamePrefix: string): string {
```

Comment:
we cannot assume that only 1 version of snap is in our local files, as for testing the snap upgrade flow, we have 2 snap versions on the webpack snap.
So in all cases we'll use the latest version (which is the only one) except if we specify otherwise (in the case of wanting to use the webpack old snap)